internal/devbox/packages.go

```diff
@@ -107,6 +107,10 @@ func (d *Devbox) Add(ctx context.Context, pkgsNames []string, opts devopt.AddOpt
 		return err
 	}
 
+	if err := d.lockfile.Add(addedPackageNames...); err != nil {
+		return err
+	}
```

Comment:
Main issue with this is that this saves the lock file right away. This will leave it in a bad state if something fails later on. It's also not 100% correct (see `newFlakePlan` calls `Add`)

A few alternatives:

* Instead of doing lockfile.Add just do `lockfile.UpdateSystemsFormat()`  in the 3 places you need it (add/remove/update). This is simplest.
* In lockfile.Save() only update the format is stuff is dirty (i.e if reloading the lockfile doesn't match what is in memory). (though it may be a bit complex to tell if dirty). This is cleanest solution because it doesn't require a new bool we may forget to add.
* Remove `Save` from `Add` (this is probably OK, but it feels risky). I don't love this one.

FWIW, currently `newFlakePlan` calls `lockfile.Add` on includes every time the state needs to be recomputed, so I think any project with includes will automatically update.

Reply:
ugh, I thought I had checked for existing callers, but I think I missed this one...good catch